app/storage/slice/state.ts

```diff
@@ -1,11 +1,21 @@
+import type { ToastStatus } from "~app/component/toast"
 import type { State } from "~storage/local/state"
 
 import type { BackgroundStateCreator } from "../middleware/background"
 
 export interface StateSlice {
   state: State
+  setToast: (message: string, status: ToastStatus) => Promise<void>
```

Comment:
I feel that toast is a transient status for app only, which would not need to persist in storage for a long time. It could be better to use toast as context   provider in order to trigger it across components.

Reply:
Good point. Now I understand why we both have context provider and storage. tks!

[create toastContext and use proivder instead of storage](https://github.com/pilagod/waallet-extension/pull/158/commits/35b1e862fb17d42f4cc9c4fc4fd9c1057a3e4680)

[import and setup ToastContext](https://github.com/pilagod/waallet-extension/pull/158/commits/754e47b5da36530d359fc2891a7cf4bc425d82db)